packages/front-end/components/Features/FeatureModal/EnvironmentSelect.tsx

```diff
@@ -12,34 +13,66 @@ const EnvironmentSelect: FC<{
 }> = ({ environmentSettings, environments, setValue }) => {
   const permissionsUtil = usePermissionsUtil();
   const { project } = useDefinitions();
+  const selectAllChecked = Object.values(environmentSettings).every(
+    (env) => env.enabled
+  );
+  const selectAllIndeterminate = Object.values(environmentSettings).some(
+    (env) => env.enabled
+  );
+
   return (
     <div className="form-group">
-      <label>Enabled Environments</label>
-      <div className="appbox px-3 pt-3 pb-2 bg-light">
-        <div className="row">
+      <Container
+        p="5"
+        style={{
+          background: "var(--color-background)",
+          borderRadius: "var(--radius-2)",
+        }}
+      >
+        <Text as="label" weight="bold" mb="4">
+          Enabled Environments
+        </Text>
+        <div>
+          <Checkbox
+            value={
+              selectAllChecked
+                ? true
+                : selectAllIndeterminate
+                ? "indeterminate"
+                : false
+            }
+            setValue={(v) =>
+              environments.forEach((env) => {
```

Comment:
issue: If I have environments that I don't have permission to create feature in, this toggle still updates it in the UI. I have not tested how the server would handle it though.

Reply:
Let me know if this latest change fixes that. Permissions are not working correctly for my test user for some reason so I can't check myself

Reply:
Was able to fix my test user and seems to be working